packages/cart/src/services/cart-module.ts

```diff
@@ -176,10 +191,378 @@ export default class CartModuleService implements ICartModuleService {
+      filters,
+      config,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<
+      CartTypes.CartShippingMethodDTO[]
+    >(methods, {
+      populate: true,
+    })
+  }
+
+  addLineItems(
+    data: CartTypes.AddLineItemsDTO,
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]>
+  addLineItems(
+    data: CartTypes.AddLineItemsDTO[],
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]>
+  addLineItems(
+    cartId: string,
+    items: CartTypes.CreateLineItemDTO[],
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]>
+
+  @InjectManager("baseRepository_")
+  async addLineItems(
+    cartIdOrData:
+      | string
+      | CartTypes.AddLineItemsDTO[]
+      | CartTypes.AddLineItemsDTO,
+    dataOrSharedContext?: CartTypes.CreateLineItemDTO[] | Context,
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]> {
+    if (isString(cartIdOrData)) {
+      return await this.addLineItems_(
+        cartIdOrData,
+        dataOrSharedContext as CartTypes.CreateLineItemDTO[],
+        sharedContext
+      )
+    }
+
+    if (Array.isArray(cartIdOrData)) {
+      return await this.addLineItemsBulk_(
+        cartIdOrData,
+        dataOrSharedContext as Context
+      )
+    }
+
+    return await this.addLineItemsBulk_(
+      [cartIdOrData],
+      dataOrSharedContext as Context
+    )
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  protected async addLineItems_(
+    cartId: string,
+    data: CartTypes.CreateLineItemDTO[],
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]> {
+    return await this.addLineItemsBulk_(
+      [{ cart_id: cartId, items: data }],
+      sharedContext
+    )
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  protected async addLineItemsBulk_(
+    data: CartTypes.AddLineItemsDTO[],
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]> {
+    const lineItemsMap = new Map<string, CartTypes.CreateLineItemDTO[]>()
+
+    for (const lineItemData of data) {
+      lineItemsMap.set(lineItemData.cart_id, lineItemData.items)
+    }
+
+    const items = await promiseAll(
+      [...lineItemsMap].map(async ([cartId, lineItems]) => {
+        return await this.lineItemService_.create(
+          cartId,
+          lineItems,
+          sharedContext
+        )
+      })
+    )
+
+    return await this.listLineItems(
+      { id: items.flat().map((c) => c.id) },
+      {},
+      sharedContext
+    )
+  }
+
+  updateLineItems(
+    data: CartTypes.UpdateLineItemsDTO,
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]>
+  updateLineItems(
+    data: CartTypes.UpdateLineItemsDTO[],
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]>
+  updateLineItems(
+    cartId: string,
+    data: CartTypes.UpdateLineItemDTO[],
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]>
+
+  @InjectManager("baseRepository_")
```

Comment:
I ll stop the review of the line items since I remembered there is a dedicated pr 😄